pkg/oci/block/block.go

```diff
@@ -30,50 +31,59 @@ const (
 	// FIXME: Assume lun 1 for now?? Can we get the LUN via the API?
 	diskIDByPathTemplate = "/dev/disk/by-path/ip-%s:%d-iscsi-%s-lun-1"
 	volumeOCIDTemplate   = "ocid1.volume.oc1.%s.%s"
+	ocidVolPrefix        = "ocid1.volume."
+	ocidPrefix           = "ocid1."
 )
 
 // OCIFlexvolumeDriver implements the flexvolume.Driver interface for OCI.
 type OCIFlexvolumeDriver struct{}
 
-// GetDriverDirectory gets the ath for the flexvolume driver either from the
-// env or default.
-func GetDriverDirectory() string {
-	// TODO(apryde): Document this ENV var.
-	path := os.Getenv("OCI_FLEXD_DRIVER_DIRECTORY")
-	if path == "" {
-		path = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/oracle~oci"
-	}
-	return path
-}
-
-// GetConfigPath gets the path to the OCI API credentials.
-func GetConfigPath() string {
-	return GetDriverDirectory() + "/config.yaml"
+// Name returns the name of the driver
+func (d OCIFlexvolumeDriver) Name() string {
+	return "oci-block"
 }
 
 // Init checks that we have the appropriate credentials and metadata API access
 // on driver initialisation.
-func (d OCIFlexvolumeDriver) Init() flexvolume.DriverStatus {
-	path := GetConfigPath()
+func (d OCIFlexvolumeDriver) Init() error {
+	path := common.GetConfigPath()
```

Comment:
Could move the body of the this into a 'checkCreds' function in common, as this could then be called from Init in both the block and filesystem drivers.